src/Compilers/VisualBasic/Portable/Symbols/ArrayTypeSymbol.vb

```diff
@@ -578,6 +584,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     Return True
                 End Get
             End Property
+
+            Friend Overrides Function WithElementType(newElementType As TypeSymbol) As ArrayTypeSymbol
+                Return New MDArray(newElementType, CustomModifiers, Rank, BaseTypeNoUseSiteDiagnostics)
+            End Function
         End Class
```

Comment:
Consider adding `MDArrayNoSizesOrBounds` and making `MDArray` abstract, so there are only abstract and sealed types, to match C#.